src/App.tsx

```diff
@@ -1,14 +1,56 @@
-/* eslint-disable max-len */
-import React from 'react';
+import React, {
+  useCallback, useEffect, useMemo, useState,
+} from 'react';
 import 'bulma/css/bulma.css';
 import '@fortawesome/fontawesome-free/css/all.css';
 
 import { TodoList } from './components/TodoList';
 import { TodoFilter } from './components/TodoFilter';
-import { TodoModal } from './components/TodoModal';
 import { Loader } from './components/Loader';
+import { Todo } from './types/Todo';
+import { getTodos } from './api';
+import { getFilteredTodos } from './utils/helpers';
+import { ALL, DEFAULT_TODO_ID } from './utils/constants';
+import { TodoModal } from './components/TodoModal';
 
 export const App: React.FC = () => {
+  const [todos, setTodos] = useState<Todo[]>([]);
+  const [error, setError] = useState(false);
+  const [isLoading, setIsLoading] = useState(false);
+  const [query, setQuery] = useState('');
+  const [sortType, setSortType] = useState(ALL);
+  const [selectedTodoById, setSelectedTodoById] = useState(DEFAULT_TODO_ID);
+
+  const visibleTodos = getFilteredTodos(todos, sortType, query);
+
+  const fetchTodos = async () => {
+    setIsLoading(true);
+
+    try {
+      setTodos(await getTodos());
+    } catch {
+      setError(true);
+    } finally {
+      setIsLoading(false);
+    }
+  };
+
+  useEffect(() => {
+    fetchTodos();
+  }, []);
+
+  const selectedTodo = useMemo(() => (
+    todos.find(({ id }): boolean => id === selectedTodoById)
+  ), [selectedTodoById]);
```

Comment:
no need to wrap it into useMemo hook
it's not a hard calculation / operation, so useMemo will only make it worse